### ISSUE TITLE
Not compatible with Firefox Quantum

### ISSUE THREAD

**Issue by vic4key** [Infocatcher/Private_Tab_WE#1]
Hi @Infocatcher,

This add-on is now not compatible with Firefox Quantum.
Could you please spend time to update it as soon as possible?
I very like your add-on, it's very useful.
Thanks you so much.

Best Regards,
Vic P.

**Comment by ghost** [comment]
Look at the README, there are 2 blocking bugs on Firefox's side. Not much the author can do unless they contribute directly to firefox itself.

**Comment by vic4key** [comment]
Oops. Hope to be able to find a good solution for this. Thanks.

**Comment by anewuser** [comment]
Check this out: https://github.com/stoically/firefox-add-on-temporary-containers

**Comment by Infocatcher** [comment]
> Check this out: https://github.com/stoically/firefox-add-on-temporary-containers

Hmm... as I see, still used something custom to remove history entries.
Store each visited URL:
https://github.com/stoically/firefox-add-on-temporary-containers/blob/ea04a06af2d40d04e4f0c6960e1e6761cedc6f05/src/background/request.js#L33-L44
And then remove them:
https://github.com/stoically/firefox-add-on-temporary-containers/blob/ea04a06af2d40d04e4f0c6960e1e6761cedc6f05/src/background/container.js#L227-L237
Better, than nothing, but also will be removed regular history, if will be opened the same URL in non-private and "private" mode/container. :(

**Comment by anewuser** [comment]
That's because of https://bugzilla.mozilla.org/show_bug.cgi?id=1283320. 

Apart from than that, he's added many combinations of options and has plans for adding more. Maybe you should collaborate with him with your ideas and experience to make it as efficient as possible instead of having to reinvent the wheel?

**Comment by vic4key** [comment]
Seem it cleans sth. Not able to use normal mode and private together.

**Comment by anewuser** [comment]
@vic4key Disable the automatic mode in the settings.

**Comment by vic4key** [comment]
Great. Thank you so much, bro.

**Comment by stoically** [comment]
@Infocatcher 

> Better, than nothing, but also will be removed regular history, if will be opened the same URL in non-private and "private" mode/container. :(

Yeah, it's also explicitly mentioned as a big red warning in the opt-in advanced preferences that exactly that happens until history support for containers is added.

Also, if you're interested, I'd be absolutely open to collaborate. Just let me know if I can be of any help if you consider starting the WebExtension Development.

@vic4key Sorry for the hassle, I'll try to make the on-boarding easier.

Thanks @anewuser for helping out!
